Kitodo/src/main/java/org/kitodo/model/Subfolder.java

```diff
@@ -0,0 +1,301 @@
+            int firstStar) {
+        String realPath = variableReplacer.replace(folder.getPath().substring(0, lastSeparator));
+        String processId = process.getId().toString();
+        final URI directory = (realPath.isEmpty() ? Paths.get(ConfigCore.getKitodoDataDirectory(), processId)
+                : Paths.get(ConfigCore.getKitodoDataDirectory(), processId, realPath)).toUri();
+        StringBuilder patternBuilder = new StringBuilder();
+        if (firstStar > 0) {
+            patternBuilder.append(Pattern.quote(lastSegment.substring(0, firstStar)));
+        }
+        patternBuilder.append("(.*)");
+        if (firstStar < lastSegment.length() - 1) {
+            patternBuilder.append(Pattern.quote(
+                lastSegment.substring(firstStar + 1).replaceFirst("\\*$", getFileFormat().getExtension(false))));
+        }
+        return Pair.of(directory, Pattern.compile(patternBuilder.toString()));
+    }
+
+    /**
+     * Calculates the search pattern in the ordinary case that the path is
+     * simply a path.
+     * 
+     * @return search request consisting of an indication of the folder to be
+     *         searched and a pattern to which the file names must correspond
+     */
+    private Pair<URI, Pattern> determineCommonSearchParameters() {
+        String processId = process.getId().toString();
+        URI directory = Paths
+                .get(ConfigCore.getKitodoDataDirectory(), processId, variableReplacer.replace(folder.getPath()))
+                .toUri();
+        String pattern = "(.*)\\." + Pattern.quote(getFileFormat().getExtension(false));
+        return Pair.of(directory, Pattern.compile(pattern));
+    }
+
+    /**
+     * Determines the directory to search and the search pattern.
+     *
+     * @return search request consisting of an indication of the folder to be
+     *         searched and a pattern to which the file names must correspond
+     */
+    private Pair<URI, Pattern> determineDirectoryAndFileNamePattern() {
+        int lastSeparator = folder.getPath().lastIndexOf(File.separatorChar);
+        String lastSegment = folder.getPath().substring(lastSeparator + 1);
+        int firstStar = lastSegment.indexOf('*');
+        if (firstStar == -1) {
+            return determineCommonSearchParameters();
+        } else {
+            return determineComplicatedSearchParameters(lastSeparator, lastSegment, firstStar);
+        }
+    }
+
+    /**
+     * Returns a file format by its MIME type, if any.
+     */
+    public FileFormat getFileFormat() {
+        try {
+            return FileFormatsConfig.getFileFormat(folder.getMimeType()).get();
+        } catch (JAXBException e) {
+            throw new UndeclaredThrowableException(e);
+        }
+    }
+
+    /**
+     * Returns the settings of the subfolder configured in the project.
+     * 
+     * @return the folder
+     */
+    public Folder getFolder() {
```

Comment:
Is it a Folder or is it settings like stated in the javadoc? Maybe the name of the bean is missleading?

Reply:
Both, in a sense. One is a folder at the core level (and in this sense the parameters of the physical folder), the other is a physical folder on the disk. It's a bit like newspapers. On the one hand there is the newspaper as a conceptual unit (complete edition, ie all the newspapers in a newspaper that have ever appeared). This corresponds to the folder object. Then there's the newspaper, a single one that's just outside the door. This corresponds to the subfolder object.